### ISSUE TITLE
[bundler] Bundled shell does not re-export ids used by multiple fragments; './x-app.js' does not provide an export named 'directive'

### ISSUE THREAD

**Issue by keanulee** [Polymer/tools#721]
Setup:
- Shell - my-app.js
- Multiple fragments - my-view1.js and my-view2.js

If my-view1 uses a directive (e.g. `repeat`):
```js
import { repeat } from 'lit-html/directives/repeat.js';
```

And my-view2 uses another directive (e.g. `unsafeHTML`):
```js
import { unsafeHTML } from 'lit-html/directives/unsafe-html.js';
```

Then both fragments will try to import `directive` from my-app, but my-app doesn't export it.

Repro:

```bash
$ git clone -b lit-element https://github.com/Polymer/shop.git
$ cd shop
$ npm i
$ polymer build
$ polymer serve build/esm-bundled
```

Navigate to the detail view (e.g. http://localhost:8081/detail/ladies_outerwear/Ladies+Pullover+L+S+Hood). Note the console error

```
shop-app.js:483 Uncaught (in promise) SyntaxError: The requested module './shop-app.js' does not provide an export named 'directive'
```

**Comment by LarsDenBakker** [comment]
See https://github.com/Polymer/lit-html/issues/515 for a minimal reproduction.

I dont see this happen for all shared imports. Possibly it has something to do with the export *

**Comment by Westbrook** [comment]
Also seeing this, any thoughts towards a fix or areas I might support investigating towards one?

**Comment by Westbrook** [comment]
Indeed @LarsDenBakker it does seem to be somehow related to the `export *` in that when you update this line in `lit-html` https://github.com/Polymer/lit-html/blob/master/src/lit-html.ts#L26 to be:
```
export { directive, isDirective } from './lib/directive.js';
```
The error is avoided.

**Comment by usergenic** [comment]
There is a bug in rollup specifically related to the use of `export * from` (apparently, only where referencing an external module, i.e. not part of the bundle) that I filed recently when tracking this down: https://github.com/rollup/rollup/issues/2489

I'm looking at possible workarounds for this within bundler.  One of them is essentially what you demonstrated, which is performing a translation of the `export * from` to an explicit named `export {a, b, c} from` as part of a before-I-hand-it-off-to-rollup step.  I may have a new PR within a couple of days or just include it as part of https://github.com/Polymer/tools/pull/722.

**Comment by Westbrook** [comment]
@usergenic were you able to get the workaround you envisioned into #722? And, do you have an idea when it might be part of a release? Unless I'm doing something wrong, which is always possible, neither versions 1.8.1 nor 1.9.0-pre.2 of `polymer-cli` seem to be building this correctly. Let me know if there is anything I can do to support getting this into the world!

**Comment by LarsDenBakker** [comment]
Can we get a release? This is a really nasty bug that we'd really like to have a fix for...

**Comment by keanulee** [comment]
Fixed in polymer-cli 1.9.0
